exercises/practice/ledger/.meta/example.cairo

```diff
@@ -0,0 +1,225 @@
+// Refactored code
+use core::to_byte_array::{AppendFormattedToByteArray};
+
+#[derive(Debug, PartialEq, Drop)]
+pub enum Currency {
+    USD,
+    EUR,
+}
+
+#[derive(Debug, PartialEq, Drop)]
+pub enum Locale {
+    en_US,
+    nl_NL,
+}
+
+pub fn format_entries(
+    currency: Currency, locale: Locale, transactions: Array<(ByteArray, ByteArray, ByteArray)>
+) -> Array<ByteArray> {
+    let mut ledger: Array<ByteArray> = ArrayTrait::new();
+
+    // Step 1: Define the header based on the locale
+    let header = match @locale {
+        Locale::en_US => "Date       | Description               | Change       ",
+        Locale::nl_NL => "Datum      | Omschrijving              | Verandering  ",
+    };
+    ledger.append(header);
+
+    // Step 2: Process transactions
+    for (
+        date, transaction, change
+    ) in transactions {
+        // Format the date based on the locale
+        let formatted_date = format_date(@date, @locale);
+
+        // Format the change based on the currency and locale
+        let formatted_change = format_change(@change, @currency, @locale);
+
+        // Format the transaction row
+        let mut row = formatted_date;
+        row += " | ";
+        row += format_transaction(transaction);
+        row += " | ";
+        row += formatted_change;
```

Comment:
You can simplify this body further:

```rust
        let date = format_date(@date, @locale);
        let change = format_change(@change, @currency, @locale);
        let row = format!("{} | {} | {}", date, format_transaction(transaction), change);
        ledger.append(row);
```

Additionally, you could move the body to a helper function `format_row` (including parameter renaming):

```rust
for transaction in transactions {
    let row = format_row(@currency, @locale, transaction);
    ledger.append(row);
};

// ...

// the inline attribute reduces gas cost a bit in this case
#[inline(always)]
fn format_row(
    currency: @Currency, locale: @Locale, transaction: (ByteArray, ByteArray, ByteArray)
) -> ByteArray {
    let (date, description, amount_in_cents) = transaction;
    let date = format_date(@date, locale);
    let amount_in_cents = format_amount(@amount_in_cents, currency, locale);
    format!("{} | {} | {}", date, format_description(description), amount_in_cents)
}
```